### PR TITLE
Reader: Follow conversation should handle `success: false`

### LINKED ISSUE #15642: Follow conversation: Handle a success:false API response
The API call to follow a conversation in the Reader can return `success:false` as part of its response.  One way to replicate this is by setting WordPress.com's [Reader Subscription notification settings](https://wordpress.com/me/notifications/subscriptions) to block all emails, and then try to follow a conversation. 

Currently, the relevant code in the app does not inspect the data returned from the API and assumes success.  This can lead to some confusion where it looks like a conversation is being followed, but tapping away from that post's comment list, and then back will show that the conversation is not followed.

Let's figure out the difference scenarios where success:false can be returned.  Then let's update the relevant code to check the response.  If the request says `success:false` let's prompt the user as appropriate.  Perhaps nudging them to check settings to see if they are blocking email.

### PR DESCRIPTION
**Pull request by mindgraffiti**
Ref. #15642 

In this PR
- the taskSucceeded bool continues to bubble up the success blocks until it reaches one where user feedback is handled
- if the user fails to follow the conversation, a toast message appears with a generic error message and reverts the Follow conversation button

**Not implemented:** better error messaging to the user. Also not implemented is a nudge for users to check their Reader subscription notification settings.  

### To test:
1. Log in with a non-a8c WordPress account, with a validated email address. 
2. On the web, disable email notifications by visiting [Reader subscription notification settings](https://wordpress.com/me/notifications/subscriptions) and checking the box in the "Block emails" section.
2. Go to Reader > Discover and find a blog post that has the comment button enabled
3. Tap the comment button, then tap the "Follow conversation by email" button
4. A toast message will appear, "Unable to subscribe to the comments" and the follow conversation button will revert from "**Following** conversation by email" to "**Follow** conversation..."

(A user will always be allowed to unfollow a conversation.)

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Reader/ReaderCommentsViewController.m [hunk 1]
```diff
@@ -1215,15 +1215,28 @@ - (void)handleFollowConversationButtonTapped
...
-            : NSLocalizedString(@"Successfully unsubscribed from the comments", @"The app successfully unsubscribed from the comments for the post");
-
-        dispatch_async(dispatch_get_main_queue(), ^{
-            [generator notificationOccurred:UINotificationFeedbackTypeSuccess];
-            [weakSelf displayNoticeWithTitle:title message:nil];
-        });
+    void (^successBlock)(BOOL taskSucceeded) = ^void(BOOL taskSucceeded) {
+        if (taskSucceeded == NO) {
+            NSString *title = newIsSubscribed
+                ? NSLocalizedString(@"Unable to subscribe to the comments", @"The app failed to subscribe to the comments for the post")
```
Comment (ScoutHarris) [inline WordPress/Classes/ViewRelated/Reader/ReaderCommentsViewController.m]: I wonder if these messages should be more in line with the button text, and other toast messages. Maybe like `Unable to follow conversation`, `Successfully followed conversation`, etc?

File: WordPress/Classes/ViewRelated/Reader/ReaderCommentsViewController.m [hunk 2]
```diff
@@ -1215,15 +1215,28 @@ - (void)handleFollowConversationButtonTapped
...
-
-        dispatch_async(dispatch_get_main_queue(), ^{
-            [generator notificationOccurred:UINotificationFeedbackTypeSuccess];
-            [weakSelf displayNoticeWithTitle:title message:nil];
-        });
+    void (^successBlock)(BOOL taskSucceeded) = ^void(BOOL taskSucceeded) {
+        if (taskSucceeded == NO) {
+            NSString *title = newIsSubscribed
+                ? NSLocalizedString(@"Unable to follow conversation", @"The app failed to subscribe to the comments for the post")
+                : NSLocalizedString(@"Failed to unfollow conversation", @"The app failed to unsubscribe from the comments for the post");
```
Comment (mindgraffiti) [inline WordPress/Classes/ViewRelated/Reader/ReaderCommentsViewController.m:1222]: @ScoutHarris this is not the final form of the error handling or the error messaging. I will be adding a second PR for that.

Comment (ScoutHarris) [reply WordPress/Classes/ViewRelated/Reader/ReaderCommentsViewController.m:1222]: Ah gotcha!

**Review by ScoutHarris — CHANGES_REQUESTED** [review]
Hey @mindgraffiti .

The button state isn't updated correctly when unfollowing a conversation.

- Uncheck 'block emails'.
- Go to Reader > Discover and find a blog post that has the comment button enabled.
- Tap the comment button, then tap the "Follow conversation by email" button.
- See toast message.
- Tap the "Following conversation by email" button" to unfollow.
- I get the unfollowed toast message, but the button shows 'Follow' for a second then changes back to 'Following'.


https://user-images.githubusercontent.com/1816888/109050177-5816e700-7696-11eb-90b2-9efe7c45c551.mp4

**Comment by mindgraffiti** [comment]
Nice catch. As soon as I re-read my success block I saw the error 🤦‍♀️ . Thank you for testing more than the described states!

**Review by ScoutHarris — APPROVED** [review]
LGTM!

:shipit:

**Pull request merged**